engine/common/requester/engine.go

```diff
@@ -426,6 +439,15 @@ func (e *Engine) onEntityResponse(originID flow.Identifier, res *messages.Entity
 		delete(needed, entityID)
 		delete(e.items, entityID)
 
+		if item.checkIntegrity {
+			actualEntityID := entity.ID()
+			// validate that we got correct entity, exactly what we were expecting
+			if entityID != actualEntityID {
+				return engine.NewInvalidInputErrorf("could not validate response body for requested entity, expected: %v, actual: %v",
```

Comment:
Error'ing out here raises a few fundamental concerns:
* We process all entities until we hit the first error and then discard the remaining ones, even if they are valid. In my mind, it would be more consistent to either accept all valid entities and discard the invalid ones (irrespective of order); or discard all entities if _at least one_ is invalid. 
* This error will escalate to the logic which invoked `Process`, i.e.  the networking layer. What is the networking layer supposed to do with the error? (it will just log it and ignore it) 

I think a log message here together with a `continue` would be fine.

Reply:
I am against returning an error. I would log error and continue

Reply:
at least for now, later when we will have slashing, we can better handle this case

Reply:
> I am against returning an error. I would log error and continue

I meant to suggest exactly the same: log and continue

```suggestion
				e.log.Error().
					Hex("origin", logging.ID(originID)).
					Hex("stated_entity_id", logging.ID(entityID)).
					Hex("provided_entity", logging.ID(actualEntityID)).
					Msg("provided entity does not match stated ID")
				continue
```